### PR TITLE
Remove exception when deleting files

### PR DESCRIPTION
**Pull request by jelliottp**
Based on https://github.com/abapGit/abapGit/issues/6710

Remove exception `Object should be removed, but has NO deletion flag in transport`. Straightaway remove objects, since they can be removed from a repo by either deleting them or by moving them to an outside package.

### CODE REVIEW

File: src/cts/zcl_abapgit_transport_objects.clas.abap [hunk 1]
```diff
@@ -1,24 +1,24 @@
...
+        io_stage           TYPE REF TO zcl_abapgit_stage
+        is_stage_objects   TYPE zif_abapgit_definitions=>ty_stage_files
+        it_object_statuses TYPE zif_abapgit_definitions=>ty_results_tt
       RAISING
-        zcx_abapgit_exception .
+        zcx_abapgit_exception.
   PROTECTED SECTION.
   PRIVATE SECTION.
 
-    DATA mt_transport_objects TYPE zif_abapgit_definitions=>ty_tadir_tt .
```
Comment (larshp) [inline src/cts/zcl_abapgit_transport_objects.clas.abap:21]: please dont change this part, it makes it more difficult to review the PR

File: src/cts/zcl_abapgit_transport_objects.clas.testclasses.abap [hunk 2]
```diff
@@ -168,7 +165,6 @@ CLASS ltcl_transport_objects IMPLEMENTATION.
   ENDMETHOD.
 
   METHOD object_not_in_local_files.
-
```
Comment (larshp) [inline src/cts/zcl_abapgit_transport_objects.clas.testclasses.abap:171]: no need to change this line, just keep it as is?

File: src/cts/zcl_abapgit_transport_objects.clas.testclasses.abap [hunk 3]
```diff
@@ -429,7 +345,6 @@ CLASS ltcl_transport_objects IMPLEMENTATION.
   ENDMETHOD.
 
   METHOD then_it_should_remove_at_stage.
-
```
Comment (larshp) [inline src/cts/zcl_abapgit_transport_objects.clas.testclasses.abap:432]: no need to change this line, just keep it as is?

File: src/cts/zcl_abapgit_transport_objects.clas.abap (hunk 1 above)
Comment (jelliottp) [reply src/cts/zcl_abapgit_transport_objects.clas.abap:21]: Ok, yeah I have some abap-cleaner settings for these things.

Comment (larshp) [reply src/cts/zcl_abapgit_transport_objects.clas.abap]: dont get me started about abap-cleaner...

Comment (jelliottp) [reply src/cts/zcl_abapgit_transport_objects.clas.abap]: I would like to hear, maybe in another forum :)

**Review by larshp — APPROVED** [review]
thanks, looks good

**Comment by larshp** [comment]
welcome to abapGit as a contributor 🎉

**Pull request merged**